tst/EnergyPlus/unit/SurfaceGeometry.unit.cc

```diff
@@ -9058,6 +9054,2395 @@ TEST_F(EnergyPlusFixture, GetSurfaceData_SurfaceOrder)
     EXPECT_FALSE(std::find(HTKivaSurfaces.begin(), HTKivaSurfaces.end(), thisSurface) != HTKivaSurfaces.end());
 }
 
+TEST_F(EnergyPlusFixture, GetSurfaceData_SurfaceOrder2)
```

Comment:
Added three new unit tests that are subsets of `GetSurfaceData_SurfaceOrder` where the first zone/space has no surfaces (`GetSurfaceData_SurfaceOrder2`), the middle zone has no surfaces (`GetSurfaceData_SurfaceOrder3`), and the last zone has no surfaces (`GetSurfaceData_SurfaceOrder4`).